ibis/expr/analysis.py

```diff
@@ -391,6 +268,14 @@ def __init__(self, parent, proj_exprs):
     def get_result(self):
         roots = find_immediate_parent_tables(self.parent.op())
         first_root = roots[0]
+        parent_op = self.parent.op()
+
+        # reprojection of the same selections
```

Comment:
This is a new optimization actually, thanks to this we eliminate redundant projections.